### ISSUE TITLE
Parameters are different when a request is retried

### ISSUE THREAD

**Issue by vskbjrn** [flexirest/flexirest#165]
### Expected behaviour

When an after_request callback is used to retry a request, the parameters yielded to a defaults proc are not the same in the second request as in the first.

### Desired behaviour

When an after_request callback is used to retry a request, the same parameters should be yielded to a defaults proc.

### How to reproduce

1. Clone the Flexirest repo (`git clone https://github.com/flexirest/flexirest.git`)
2. Change to the flexirest folder (`cd flexirest`)
3. Install any dependencies (`bundle install`)
4. Run an interactive console (`rake c`)
5. Paste the following:

```ruby
class MyObject < Flexirest::Base
  base_url "https://ensnvmzi8hr4e.x.pipedream.net"
  get :all, "/objects", defaults: proc { |params| { type: params.delete(:object_type) } }

  after_request :try_again

  private

  def try_again(name, response)
    raise Flexirest::CallbackRetryRequestException.new
  end
end

MyObject.all(object_type: 'foo')
```

Log output:
```
  Flexirest MyObject#all - Requesting https://ensnvmzi8hr4e.x.pipedream.net/objects?type=foo
  Flexirest MyObject#all - Requesting https://ensnvmzi8hr4e.x.pipedream.net/objects?type=
  Flexirest MyObject#all - Response received 16 bytes
  Flexirest (432.5ms) MyObject#all
  Flexirest (872.4ms) MyObject#all
```

**Comment by pgib** [comment]
The `params` that are [passed in](https://github.com/flexirest/flexirest/blob/f5ed00508831ceb10d6c24b14da5f564058493f2/lib/flexirest/request.rb#L354) would have to be cloned. You're removing the key you want from the parameters with `params.delete`, so after the first attempt, it no longer exists.

**Comment by andyjeffries** [comment]
This should be fixed with version 1.12.2.
